File: backEnd/Systetica_Api/src/main/java/com/frfs/systetica/entity/Usuario.java
package com.frfs.systetica.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.sun.istack.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "usuario")
@EqualsAndHashCode
public class Usuario implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(name = "nome", length = 100)
    private String nome;

    @NotNull
    @Column(name = "telefone", length = 15)
    private String telefone;

    @NotNull
    @Column(name = "email", unique = true, length = 100)
    private String email;

    @JsonProperty(value = "password", access = JsonProperty.Access.WRITE_ONLY)
    @Column(name = "password", nullable = false, length = 100)
    private String password;

    @NotNull
    @Column(name = "codigo_aleatorio")
    private Integer codigoAleatorio;

    @Column(name = "data_codigo")
    private Date dataCodigo;

    @Column(name = "usuario_ativo")
    private Boolean usuarioAtivo;

    @NotNull
    @Column(name = "data_cadastro")
    private Date dataCadastro;

    @ManyToMany(fetch = FetchType.EAGER)
    private Collection<Role> roles = new ArrayList<>();

    @Lob
    @Column(name = "imagem_base64", length = 16777215)
    private String imagemBase64;

    @Column(name = "permissao_funcionario")
    private Boolean permissaoFuncionario;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "ass_funcionario_empresa",
            uniqueConstraints = @UniqueConstraint(columnNames = {"id_usuario", "id_empresa"}),
            joinColumns = @JoinColumn(name = "id_usuario"),
            inverseJoinColumns = @JoinColumn(name = "id_empresa"))
    private List<Empresa> empresas = new ArrayList<>();

    public List<Empresa> getEmpresas() {
        return null;
    }
}

File: frontEnd/Systetica_App/lib/model/Usuario.dart
// ignore_for_file: file_names

import 'package:systetica/model/Role.dart';

class Usuario {
  Usuario({
    this.id,
    this.nome,
    this.telefone,
    this.email,
    this.password,
    this.codigoAleatorio,
    this.roles,
    this.imagemBase64,
    this.permissaoFuncionario,
    this.emailAdministrativo,
  });

  int? id;
  String? nome;
  String? telefone;
  String? email;
  String? password;
  int? codigoAleatorio;
  List<Role>? roles;
  String? imagemBase64;
  bool? permissaoFuncionario;
  String? emailAdministrativo;
  bool selecionado = false;

  Usuario.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nome = json['nome'];
    telefone = json['telefone'];
    email = json['email'];
    password = json['password'];
    codigoAleatorio = json['codigo_aleatorio'];
    roles = json['roles'] != null
        ? (json['roles'].map<Role>((e) => Role.fromJson(e))).toList()
        : [];
    imagemBase64 = json['imagem_base64'];
    permissaoFuncionario = json['permissao_funcionario'];
    emailAdministrativo = json['email_administrativo'];
  }

  static List<Usuario> fromJsonList(List json) {
    return json.map((item) => Usuario.fromJson(item)).toList();
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['id'] = id ?? "";
    _data['nome'] = nome;
    _data['telefone'] = telefone;
    _data['email'] = email;
    _data['password'] = password;
    _data['password'] = password;
    _data['codigo_aleatorio'] = codigoAleatorio;
    _data['roles'] = null;
    _data['imagem_base64'] = imagemBase64;
    _data['permissao_funcionario'] = permissaoFuncionario;
    _data['email_administrativo'] = emailAdministrativo;
    return _data;
  }
}
